actor/builtin/miner/miner.go

```diff
@@ -465,59 +497,40 @@ func (ma *Actor) CommitSector(ctx exec.VMContext, sectorID uint64, commD, commR,
 		return 1, errors.NewRevertError("invalid sized commRStar")
 	}
 
-	// As with submitPoSt messages, bootstrap miner actors don't verify
-	// the commitSector messages that they are sent.
-	//
-	// This switching will be removed when issue #2270 is completed.
-	if !ma.Bootstrap {
-		// This unfortunate environment variable-checking needs to happen because
-		// the PoRep verification operation needs to know some things (e.g. size)
-		// about the sector for which the proof was generated in order to verify.
-		//
-		// It is undefined behavior for a miner using "LiveProofsMode" to verify
-		// a proof created by a miner in "TestProofsMode"(and vice-versa).
+	var state State
+	_, err := actor.WithState(ctx, &state, func() (interface{}, error) {
+		// As with submitPoSt messages, bootstrap miner actors don't verify
+		// the commitSector messages that they are sent.
 		//
-		proofsMode, err := GetProofsMode(ctx)
-		if err != nil {
-			return ErrGetProofsModeFailed, Errors[ErrGetProofsModeFailed]
-		}
-
-		var sectorSize types.SectorSize
-		if proofsMode == types.TestProofsMode {
-			sectorSize = types.OneKiBSectorSize
-		} else {
-			sectorSize = types.TwoHundredFiftySixMiBSectorSize
-		}
-
-		req := proofs.VerifySealRequest{}
-		copy(req.CommD[:], commD)
-		copy(req.CommR[:], commR)
-		copy(req.CommRStar[:], commRStar)
-		req.Proof = proof
-		req.ProverID = sectorbuilder.AddressToProverID(ctx.Message().To)
-		req.SectorID = sectorbuilder.SectorIDToBytes(sectorID)
-		req.SectorSize = sectorSize
-
-		res, err := (&proofs.RustVerifier{}).VerifySeal(req)
-		if err != nil {
-			return 1, errors.RevertErrorWrap(err, "failed to verify seal proof")
-		}
-		if !res.IsValid {
-			return ErrInvalidSealProof, Errors[ErrInvalidSealProof]
+		// This switching will be removed when issue #2270 is completed.
+		if !ma.Bootstrap {
+			req := proofs.VerifySealRequest{}
+			copy(req.CommD[:], commD)
+			copy(req.CommR[:], commR)
+			copy(req.CommRStar[:], commRStar)
+			req.Proof = proof
+			req.ProverID = sectorbuilder.AddressToProverID(ctx.Message().To)
+			req.SectorID = sectorbuilder.SectorIDToBytes(sectorID)
+			req.SectorSize = state.SectorSize
+
+			res, err := (&proofs.RustVerifier{}).VerifySeal(req)
```

Comment:
Not your change and not blocking, but is there a reason we don't have a constructor for `RustVerifier`?

Reply:
Yeah: The struct itself is stateless. `Verifier` is an interface and `RustVerifier` satisfies the interface with a bunch of data-in, data-out CGO calls. 

Really, I should just eliminate the Java-style `Verifier` interface and `RustVerifier` struct and rewrite its methods as functions.